cirq-core/cirq/sim/simulator_base_test.py

```diff
@@ -42,6 +52,51 @@ def _act_on_fallback_(self, action: Any, allow_decompose: bool):
         self.gate_count += 1
         return True
 
+    def join(self, other: 'CountingActOnArgs') -> 'CountingActOnArgs':
```

Comment:
It's probably worthwhile to have a separate `NonSplittableCountingActOnArgs` that doesn't provide implementations of these things at all, to ensure setting `split_entangled_states=False` is all you have to do to keep it functioning.